src/HYPRE.jl

```diff
@@ -512,42 +540,49 @@ function Internals.copy_check(dst::HYPREVector, src::PVector)
     if il_dst != il_src && iu_dst != iu_src
         # TODO: Why require this?
         throw(ArgumentError(
-            "row owner mismatch between dst ($(il_dst:iu_dst)) and src ($(il_dst:iu_dst))"
+            "row owner mismatch between dst ($(il_dst:iu_dst)) and src ($(il_src:iu_src))"
         ))
     end
 end
 
 # TODO: Other eltypes could be support by using a intermediate buffer
-function Base.copy!(dst::PVector{HYPRE_Complex}, src::HYPREVector)
+function Base.copy!(dst::PVector, src::HYPREVector)
     Internals.copy_check(src, dst)
-    map_parts(dst.values, dst.owned_values, dst.rows.partition) do vv, _, vr
-        il_src_part = vr.lid_to_gid[vr.oid_to_lid.start]
-        iu_src_part = vr.lid_to_gid[vr.oid_to_lid.stop]
+    map(own_values(dst), dst.index_partition) do ov, vr
```

Comment:
I guess the SubArray issue can be circumvented by using `local_values(dst)` here, right?

Reply:
I think so. I was not sure if `HYPRE_IJVectorGetValues` would safely output the ghost values too.

Reply:
I was thinking the indices would just be for the owned ones, but it should be fine to essentially pass a too-long buffer to `HYPRE_IJVectorGetValues`.